### PR TITLE
Fix issue #77; Diff. meth. reports in separate html files

### LINKED ISSUE #77: Right now only one round of diffmeth is incorporated into the report
Right now only one round of diffmeth is incorporated into the report

### PR DESCRIPTION
**Pull request by katwre**
Here is implemented differential methylation output as a separate html report for each comparison. It fixes #77 issue.

Let me know if it works for you and if its oki!

Since we don't have a test data that could test pieces of code to generate diff. meth. reports, here is how I faked some diff meth. Cs:
so in report_templates/diffmeth.Rmd.in I just replaced 
```{r DiffMeth.load, eval=DiffMeth}
methylDiff.obj <- readRDS(methylDiff_file)
methylDiff.obj.hypo <- readRDS(methylDiff_hypo_file)
methylDiff.obj.hyper <- readRDS(methylDiff_hyper_file)

sampleids <- methylDiff.obj@sample.ids
treatment <- getTreatment(methylDiff.obj)
```
with 
```{r DiffMeth.load, eval=DiffMeth}
methylDiff.obj <- readRDS(methylDiff_file)
methylDiff.obj.hypo <- readRDS(methylDiff_hypo_file)
methylDiff.obj.hyper <- readRDS(methylDiff_hyper_file)

sampleids <- methylDiff.obj@sample.ids
treatment <- getTreatment(methylDiff.obj)

################################################################TODO: Remove me START
library(methylKit)
file.list=list( system.file("extdata", 
                            "test1.myCpG.txt", package = "methylKit"),
                system.file("extdata", 
                            "control2.myCpG.txt", package = "methylKit") )
# read the files to a methylRawList object: myobj
myobj=methRead(file.list,
           sample.id=list("test1","ctrl1"),
           assembly="hg18",
           treatment=c(1,0),
           context="CpG"
           )
meth=unite(myobj, destrand=FALSE)
myDiff=calculateDiffMeth(meth)

methylDiff.obj <- myDiff
methylDiff.obj.hypo <- myDiff
methylDiff.obj.hyper <- myDiff
################################################################TODO: Remove me END

```

### CODE REVIEW

File: report_templates/diffmeth.Rmd.in [hunk 1]
```diff
@@ -0,0 +1,371 @@
...
+    number_sections: FALSE
+    code_folding: "hide"
+    self_contained: TRUE
+    includes:
+      in_header: pigx_bsseq_logo.html
+date: '`r format(as.POSIXct(if ("" != Sys.getenv("SOURCE_DATE_EPOCH")) { as.numeric(Sys.getenv("SOURCE_DATE_EPOCH")) } else { Sys.time() }, origin="1970-01-01"), "%Y-%m-%d %H:%M:%S")`'
+bibliography: reports.bib
+link-citations: yes
+params:
+  MethCall: TRUE
```
Comment (alexg9010) [inline report_templates/diffmeth.Rmd.in]: This line can be removed

File: BSseq_pipeline.py [hunk 2]
```diff
@@ -585,14 +595,40 @@
...
+    params:
+        chrom_seqlengths  = os.path.join(DIR_mapped,"Refgen_"+ASSEMBLY+"_chromlengths.csv"),
+        source_dir  = config['locations']['input-dir'],
+        out_dir     = config['locations']['output-dir'],
+        diffmeth_dir = DIR_diffmeth,
+        treatment = lambda wc: str(wc.treatment).replace('vs', '_'),
+        assembly    = ASSEMBLY,
+        genome_dir  = config['locations']['genome-dir'],
+        scripts_dir = DIR_scripts,
+        CpGfile     = config['general']['differential-methylation']['annotation']['CpGfile'],
```
Comment (alexg9010) [inline BSseq_pipeline.py]: Could we change the name of the cpg-islands bed file from "CpGfile", "cpgIsland_bedfile"?

File: report_templates/diffmeth.Rmd.in [hunk 3]
```diff
@@ -0,0 +1,371 @@
...
+  source_dir: ''
+  out_dir:    ''
+  diffmeth_dir: ''
+  assembly: ''
+  DiffMeth: TRUE
+  methylDiff_files: list()
+  genome_dir: ''
+  scripts_dir: ''
+  AnnotateDiffMeth: TRUE
+  CpGfile:      ''
```
Comment (alexg9010) [inline report_templates/diffmeth.Rmd.in]: Same change here

File: report_templates/diffmeth.Rmd.in [hunk 4]
```diff
@@ -0,0 +1,371 @@
...
+knitr::opts_knit$set(progress = FALSE)
+```
+
+
+```{r chunk_eval_options}
+
+DiffMeth          <- params$DiffMeth
+AnnotateDiffMeth  <- params$AnnotateDiffMeth
+
+CpGfile           <- params$CpGfile
```
Comment (alexg9010) [inline report_templates/diffmeth.Rmd.in]: and here

File: report_templates/diffmeth.Rmd.in [hunk 5]
```diff
@@ -0,0 +1,371 @@
...
+```{r AnnotateDiffMeth.annotate_with_refseq_genes, eval=AnnotateDiffMeth}
+source(paste0(scripts_dir, "fetch_procedures.R"))
+fetched.refgenes <- lookupBedFile(type = "refGene",
+                               filename = refGenes_bedfile,
+                               assembly = assembly,
+                               webfetch = webfetch )
+fetch_refgen_success <- ( fetched.refgenes != '')
+
+fetched.cpgi <- lookupBedFile(type = "cpgIslandExt",
+                              filename =  CpGfile,
```
Comment (alexg9010) [inline report_templates/diffmeth.Rmd.in]: and here

File: BSseq_pipeline.py (hunk 2 above)
Comment (Blosberg) [reply BSseq_pipeline.py]: I agree. More descriptive names are generally desireable

**Pull request merged**